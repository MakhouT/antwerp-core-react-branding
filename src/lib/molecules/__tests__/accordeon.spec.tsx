import * as React from "react";
import * as renderer from "react-test-renderer";

import * as A from "../..";

describe("Accordeon", () => {
    it("Renders accordeon", () => {
        // Act
        const component: any = renderer.create(
            <A.Accordeon>
                <A.AccordeonItem header={"Acc 1"} />
                <A.AccordeonItem header={"Acc 2"} />
                <A.AccordeonItem header={"Acc 3"} />
            </A.Accordeon>);
        const tree: renderer.ReactTestRendererJSON = component.toJSON();

        // Assert
        expect(tree).toMatchSnapshot();
        expect(tree.children.length).toEqual(3);
    });

    it("Renders accordeon with active index", () => {
        // Act
        const component: any = renderer.create(
            <A.Accordeon activeIndex={1}>
                <A.AccordeonItem header={"Acc 1"} />
                <A.AccordeonItem header={"Acc 2"} />
                <A.AccordeonItem header={"Acc 3"} />
            </A.Accordeon>);

        const tree: renderer.ReactTestRendererJSON = component.toJSON();

        // Assert
        expect(tree).toMatchSnapshot();
        expect(tree.children.length).toEqual(3);
    });

    it("Doesn't render non AccordeonItem items", () => {
        // Act
        const component: any = renderer.create(
            <A.Accordeon activeIndex={1}>
                <A.AccordeonItem header={"Acc 1"} />
                <A.AccordeonItem header={"Acc 2"} />
                <div>Don't render me</div>
            </A.Accordeon>);

        const tree: renderer.ReactTestRendererJSON = component.toJSON();

        // Assert
        expect(tree).toMatchSnapshot();
        expect(tree.children.length).toEqual(2);
    });

    it("Renders accordeon without items", () => {
        // Act
        const component: any = renderer.create(
            <A.Accordeon activeIndex={1}>
            </A.Accordeon>);

        const tree: renderer.ReactTestRendererJSON = component.toJSON();

        // Assert
        expect(tree).toMatchSnapshot();
    });

});
