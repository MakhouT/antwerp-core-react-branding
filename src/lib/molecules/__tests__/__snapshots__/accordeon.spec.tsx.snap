// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`accordeon Doesn't render non AccordeonItem items 1`] = `
<div
  className="m-accordion"
>
  <div
    className="m-accordion__tab is"
  >
    <div
      className="m-accordion__header"
      onClick={[Function]}
    >
      Acc 1
    </div>
    <div
      className="m-accordion__content"
    >
      <div
        className="u-margin"
        style={undefined}
      />
    </div>
  </div>
  <div
    className="m-accordion__tab is-open"
  >
    <div
      className="m-accordion__header"
      onClick={[Function]}
    >
      Acc 2
    </div>
    <div
      className="m-accordion__content"
    >
      <div
        className="u-margin"
        style={undefined}
      />
    </div>
  </div>
</div>
`;

exports[`accordeon Renders accordeon 1`] = `
<div
  className="m-accordion"
>
  <div
    className="m-accordion__tab is-open"
  >
    <div
      className="m-accordion__header"
      onClick={[Function]}
    >
      Acc 1
    </div>
    <div
      className="m-accordion__content"
    >
      <div
        className="u-margin"
        style={undefined}
      />
    </div>
  </div>
  <div
    className="m-accordion__tab is"
  >
    <div
      className="m-accordion__header"
      onClick={[Function]}
    >
      Acc 2
    </div>
    <div
      className="m-accordion__content"
    >
      <div
        className="u-margin"
        style={undefined}
      />
    </div>
  </div>
  <div
    className="m-accordion__tab is"
  >
    <div
      className="m-accordion__header"
      onClick={[Function]}
    >
      Acc 3
    </div>
    <div
      className="m-accordion__content"
    >
      <div
        className="u-margin"
        style={undefined}
      />
    </div>
  </div>
</div>
`;

exports[`accordeon Renders accordeon with active index 1`] = `
<div
  className="m-accordion"
>
  <div
    className="m-accordion__tab is"
  >
    <div
      className="m-accordion__header"
      onClick={[Function]}
    >
      Acc 1
    </div>
    <div
      className="m-accordion__content"
    >
      <div
        className="u-margin"
        style={undefined}
      />
    </div>
  </div>
  <div
    className="m-accordion__tab is-open"
  >
    <div
      className="m-accordion__header"
      onClick={[Function]}
    >
      Acc 2
    </div>
    <div
      className="m-accordion__content"
    >
      <div
        className="u-margin"
        style={undefined}
      />
    </div>
  </div>
  <div
    className="m-accordion__tab is"
  >
    <div
      className="m-accordion__header"
      onClick={[Function]}
    >
      Acc 3
    </div>
    <div
      className="m-accordion__content"
    >
      <div
        className="u-margin"
        style={undefined}
      />
    </div>
  </div>
</div>
`;

exports[`accordeon Renders accordeon without items 1`] = `
<div
  className="m-accordion"
/>
`;
